src/payload.rs

```diff
@@ -126,46 +160,71 @@ impl<N> PayloadW<N>
 where
     N: ArrayLength<u8>,
 {
+    /// Update the header contained within this grant.
+    ///
+    /// This can be used to modify the pipe, length, etc. of the
+    /// packet.
+    pub fn update_header(&mut self, header: EsbHeader) {
```

Comment:
I guess this also proves that we need some method for the user to create an `EsbHeader`.